rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/topic_cache.hpp

```diff
@@ -180,6 +223,7 @@ class TopicCache
         "rmw_fastrtps_shared_cpp",
         "Unable to remove topic, does not exist '%s' with type '%s'",
         topic_name.c_str(), type_name.c_str());
+      return false;
```

Comment:
Added this since it seems like previously even a failed removal returned `true`.

Reply:
Well, if the code run up to here, it means that the topic info was at least removed from `topic_name_to_topic_data_`.
I would return `true`, but a warning/error instead of a debug message.